### ISSUE TITLE
All vikidia recipes are failing

### ISSUE THREAD

**Issue by benoit74** [openzim/zim-requests#846]
All vikidia recipes are failing in the Zimfarm: https://farm.openzim.org/recipes?name=vikidia

I investiaged a bit and there is three kind of issues:

A first group of recipes are blocked on a "bad article". This has been refused to be solved in https://github.com/openzim/mwoffliner/issues/1942 ; I don't know how to proceed from here. Recipes:
- [vikidia_ca](https://farm.openzim.org/recipes/vikidia_ca)
- [vikidia_ru](https://farm.openzim.org/recipes/vikidia_ru)
- [vikidia_fr](https://farm.openzim.org/recipes/vikidia_fr)
- [vikidia_eu](https://farm.openzim.org/recipes/vikidia_eu)
- [vikidia_es](https://farm.openzim.org/recipes/vikidia_es)
- [vikidia_en](https://farm.openzim.org/recipes/vikidia_en)

[vikidia_scn](https://farm.openzim.org/recipes/vikidia_scn) is failing with `Error parsing article .to` with an inner error `Cannot read properties of undefined (reading 'replace')`. I've opened a new upstream ticket: https://github.com/openzim/mwoffliner/issues/1998.

[vikidia_it](https://farm.openzim.org/recipes/vikidia_it) is failing with `[error] No available desktop renderer.` and no other message in the log. I've opened a new upstream ticket: https://github.com/openzim/mwoffliner/issues/1999

I've disabled all recipes for now.

**Comment by benoit74** [comment]
All their wiki are on Mediawiki 1.39.4. 

1.39 is the latest/current LTS version of Mediawiki. 1.39.4 is dated from June 30.  2023.

I would rather suspect this is a too recent version of Mediawiki for mwoflliner

**Comment by benoit74** [comment]
All recipes are now blocked by Cloudflare at the very beginning of the scrape. Might be fixed by https://github.com/kiwix/operations/issues/373 (but probably not going to happen very soon, long term issue)

**Comment by kelson42** [comment]
@benoit74 Can vikida whitelist our scraper IP? I mean technically?

**Comment by benoit74** [comment]
Things are possible in Cloudflare, but this ecosystem is quite obscure to me (e.g. is it possible to whitelist IPs on all plans or only some ...). Hence the recommendation to get approved by Cloudflare on the long term.

**Comment by kelson42** [comment]
@benoit74 Would you have a list of IP to whitelist I could transmit to Vikidia ops team?

**Comment by benoit74** [comment]
Sent through Slack

**Comment by kelson42** [comment]
I have requested the whitelisting of these IPs per email.

**Comment by kelson42** [comment]
@benoit74 The WAF has been updated and we have fixed many problematic things in MWoffliner in the meantime. But [my last attempt still fails](https://farm.openzim.org/pipeline/af45c439-8789-484f-ad54-7984440f7a82/debug). What should be fixed/investigated next (ticket number)?

**Comment by benoit74** [comment]
Looks like you stumbled upon a random failure of the mediawiki instance. [Previous run](https://farm.openzim.org/pipeline/bd28ec67-94f6-4c0d-8f45-f50c4307c57a/debug) failed due to https://github.com/openzim/mwoffliner/issues/2389 ; I did not implemented retries on 500 errors because I expected them to not be transient ... looks like this is wrong.

Let's simply restart it again for now.

**Comment by kelson42** [comment]
@benoit74 Failed again with a 500 error :(

https://farm.openzim.org/pipeline/77e84dbd-8772-4b08-9a3a-6615604610dc/debug

**Comment by benoit74** [comment]
Opened https://github.com/openzim/mwoffliner/issues/2400

**Comment by benoit74** [comment]
Upstream bug is solved, but new run did not succeeded:

https://farm.openzim.org/pipeline/1eb461ba-ad63-490c-a97d-8fa2cd5b4dd2/debug

I contacted vikidia team to ask for guidance (in French, so I'm not copy-pasting discussion here). It is disturbing to receive 503 errors and then finish with a 524.

**Comment by benoit74** [comment]
Vikidia team responded that configuration is fine on their end, and that the traffic generated by the scraper is very acceptable from their point of view. They suspected the 524 was caused by a DDoS which happened the same day. I just restarted the recipe and it failed again with a 524. I don't know what is next.

**Comment by benoit74** [comment]
Vikidia team says the 524 is a timeout by Cloudflare due to upstream (Mediawiki) timeout on big pages. They suggest we try again after 5 minutes to not overwhelm the server. Scraper is not capable to do this ATM.

**Comment by benoit74** [comment]
We've finally achieved to rebuild FR versions

@Popolechien can you please have a look at https://dev.library.kiwix.org/#lang=&q=vikidia and confirm we can move these recipes back to production? (I will assume other languages are fine)

For the record we need:
- IP + User-Agent whitelisting + skip Under Attack mode by vikidia team in Cloudflare
- retry intermittent slowness where Cloudflare returns a 524

**Comment by Popolechien** [comment]
Yep, looks good to me, thanks. I'm actually happily surprised we capture the Portal: namespace.

**Comment by kelson42** [comment]
The scrape of Vikidia FR seems really unstable and has failed twice since it has been successful in staging, see https://farm.openzim.org/recipes/vikidia_fr. I have tried again today.

**Comment by kelson42** [comment]
... and it failed... again.

@benoit74 Can younplease put the ZIM which has been created in dev in prod? Before looking to the reason why we keepmfailing to scrape Vikidia FR?

**Comment by benoit74** [comment]
Files moved to prod, will be grabbed by the library in coming hour(s). Closing the issue since this topic is closed, I will open another one for the specific `vikidia_fr` issue for clarity.
